dream_layer_frontend/src/hooks/useHistoryReducer.ts

```
@@ -0,0 +1,158 @@
+}
+
+const MAX_HISTORY_SIZE = 25;
+
+function historyReducer<T>(state: HistoryState<T>, action: HistoryAction<T>): HistoryState<T> {
+  const { past, present, future } = state;
+
+  switch (action.type) {
+    case 'UNDO':
+      if (past.length === 0) return state;
+      
+      const previous = past[past.length - 1];
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)